src/vm/dllimport.cpp

```diff
@@ -6175,11 +6217,7 @@ HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD, LoadLibErrorTracke
     for (int i = 0; hmod == NULL && i < numberOfVariations; i++)
     {
         SString currLibNameVariation;
-        currLibNameVariation.Printf(prefixSuffixCombinations[i], PAL_SHLIB_PREFIX, name, PAL_SHLIB_SUFFIX);
-#else
-    {
-        LPCWSTR currLibNameVariation = wszLibName;
-#endif
+        currLibNameVariation.Printf(prefixSuffixCombinations[i], PLATFORM_SHARED_LIB_PREFIX_A, name, PLATFORM_SHARED_LIB_SUFFIX_A);
```

Comment:
We have converted the name from UTF8 to Unicode above once already.

The PrintF is going to convert it again; but I do not think it is going to use UTF8 on Windows. I think this needs to use the converted name to Unicode (wszLibName); not the UTF8 name.

Reply:
Update all of these argument to WCHAR* instead of char*